cmd/nri-kubernetes/main.go

```diff
@@ -3,152 +3,101 @@ package main
 import (
 	"context"
 	"fmt"
+	"os"
 	"time"
 
 	"github.com/newrelic/infra-integrations-sdk/integration"
 	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/sethgrid/pester"
-	"k8s.io/apimachinery/pkg/version"
 
 	"github.com/newrelic/nri-kubernetes/v2/internal/config"
-	"github.com/newrelic/nri-kubernetes/v2/internal/discovery"
-	"github.com/newrelic/nri-kubernetes/v2/src/apiserver"
 	"github.com/newrelic/nri-kubernetes/v2/src/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/ksm"
-	"github.com/newrelic/nri-kubernetes/v2/src/metric"
-	"github.com/newrelic/nri-kubernetes/v2/src/scrape"
+	ksmClient "github.com/newrelic/nri-kubernetes/v2/src/ksm/client"
 	"github.com/newrelic/nri-kubernetes/v2/src/sink"
-	"github.com/newrelic/nri-kubernetes/v2/src/storage"
 )
 
-const defaultLabelSelector = "app.kubernetes.io/name=kube-state-metrics"
+var logger log.Logger
 
-type clientsCluster struct {
-	k8s                 client.Kubernetes
-	ksm                 ksm.Client
-	api                 apiserver.Client
-	endpointsDiscoverer discovery.EndpointsDiscoverer
-	servicesDiscoverer  discovery.ServiceDiscoverer
+const (
+	ExitClients = iota
+	ExitIntegration
+	ExitLoop
+)
+
+type clusterClients struct {
+	k8s client.Kubernetes
+	ksm ksmClient.Client
 }
 
 func main() {
 	c := config.LoadConfig()
-	logger := log.NewStdErr(c.Verbose)
+	logger = log.NewStdErr(c.Verbose)
 
-	k8s, err := client.NewKubernetes(true)
+	// TODO: Can this error?
+	conf := config.LoadConfig()
+
+	clients, err := buildClients()
 	if err != nil {
-		log.Fatal(err)
+		log.Error(err.Error())
+		os.Exit(ExitClients)
 	}
 
-	apiClient := apiserver.NewFileCacheClientWrapper(apiserver.NewClient(k8s), client.DiscoveryCacherConfig{
-		TTL:       3 * time.Hour,
-		TTLJitter: 50,
-		Storage:   storage.NewJSONDiskStorage("/var/cache/nr-kubernetes/apiserverK8SVersion"),
-	})
-
-	ksmDiscoverer, err := getDiscoverer(c, k8s, logger)
+	i, err := createIntegrationWithHTTPSink()
 	if err != nil {
-		log.Fatal(err)
+		logger.Errorf("creating integration with http sink: %w", err)
+		os.Exit(ExitIntegration)
 	}
 
-	servicesDiscoverer := discovery.NewServicesDiscoverer(k8s.GetClient())
-
-	ksmClient, err := ksm.NewKSMClient(logger)
+	// TODO: Here we will switch-case between components: KSM, ControlPlane, etc.
+	err = runKSM(&conf, clients, i)
```

Comment:
are we going to deal with routines, and all that in a different PR right?

Reply:
Yes, when we make the next change (e.g. kubelet) we'll refactor the main and add subcommands and everything. For the moment it's just a mockup :)